### ISSUE TITLE
Processes started by cross emulator can't exec other binaries (?)

### ISSUE THREAD

**Issue by sternenseemann** [NixOS/nixpkgs#119885]
**Describe the bug**

Consider the following expression where `koka` is a shell script created using `wrapProgram`:

```nix
{ pkgs ? import <nixpkgs> {} }:

with pkgs;

let
  kokaHelloWorld = writeText "hello.kk" ''
    fun main() {
      println("Hello, World!");
    }
  '';

  crossTest = crossPkgs:
    crossPkgs.runCommand "koka-cross-test" {} ''
      ${crossPkgs.stdenv.hostPlatform.emulator crossPkgs.buildPackages} \
        ${crossPkgs.koka}/bin/koka ${kokaHelloWorld} \
        > $out
    '';
in

crossTest pkgsCross.aarch64-multiplatform
```

This will fail with the following error message:

```
Error while loading /nix/store/dnv68agszwxm9rw4njsjl2di3fvvvzph-koka-2.1.1-aarch64-unknown-linux-gnu/bin/koka: Exec format error
```

At first I suspected that qemu can't deal with shebang scripts for some reason since swapping out `koka` for `.koka-wrapped`
works (at least the binary runs, but fails due to missing certain environment variables). But the issue seems to be even worse than that:

* Using `${crossPkgs.bash} -c ${crossPkgs.koka}/bin/koka` doesn't work
* Using `${crossPkgs.bash} -c ${crossPkgs.koka}/bin/.koka-wrapped` doesn't work

Seems like it's pretty much impossible for a process started by the emulator to call `exec` without running into executable format issues. Not sure if this is fixable without launching a full VM?!

Note that this makes most sense to test against #119468.

**Notify maintainers**

cc @Ericson2314 @matthewbauer @siraben

**Comment by Ericson2314** [comment]
Well, can't say I'm too surprised. I googled "binfmt_misc" and "child"/"fork" and didn't see anything obvious either.

**Comment by domenkozar** [comment]
According to [Qemu docs](https://wiki.archlinux.org/title/QEMU#QEMU_variants) we're using `qemu-${arch}` which is user emulation, but you could try with `qemu-system-${arch}` which is full emulation but slower.

I'm new to all this so I might be wrong :)

**Comment by Mindavi** [comment]
I don't think we can really solve this in nixpkgs. However, there's a new tool called makeBinaryWrapper in our toolbox which may or may not help with this situation.
